From 0d64b2043dfbfc3ab72bf9eda77c64dea9558c42 Mon Sep 17 00:00:00 2001
From: Alexandre Belloni <alexandre.belloni@free-electrons.com>
Date: Wed, 15 Jan 2014 12:00:07 +0100
Subject: [PATCH 3/3] wpa-supplicant-2.0: don't exit in pkg_postinst

Exiting explicitly in pkg_postinst makes it impossible to use the
update-rc.d class in a .bbappend because the link creation is appended
to the pkg_postinst script.

Signed-off-by: Alexandre Belloni <alexandre.belloni@free-electrons.com>
---
 meta/recipes-connectivity/wpa-supplicant/wpa-supplicant-2.0.inc | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/meta/recipes-connectivity/wpa-supplicant/wpa-supplicant-2.0.inc b/meta/recipes-connectivity/wpa-supplicant/wpa-supplicant-2.0.inc
index ce669a1ffa4f..ba28c6bd4528 100644
--- a/meta/recipes-connectivity/wpa-supplicant/wpa-supplicant-2.0.inc
+++ b/meta/recipes-connectivity/wpa-supplicant/wpa-supplicant-2.0.inc
@@ -82,9 +82,8 @@ do_install () {
 
 pkg_postinst_wpa-supplicant () {
 	# If we're offline, we don't need to do this.
-	if [ "x$D" != "x" ]; then
-		exit 0
+	if [ "x$D" = "x" ]; then
+		killall -q -HUP dbus-daemon || true
 	fi
 
-	killall -q -HUP dbus-daemon || true
 }
-- 
1.8.3.2

